### PR TITLE
[Debt] Make application doc async

### LINKED ISSUE #11126: ♻️ Make application document async
## ♻️ Debt/Refactor

Move the application document to be generated asynchronously on the server.

## 🕵️ Details

We recently completed the generation of the user profile documents but still need to do the application document there. The difference being, the application is the snapshot and includes some details about the application where the profile document is the up to date user profile without application information.

## 🙋‍♀️ Proposed Solution

Follow what was done in the [user profile document](https://github.com/GCTC-NTGC/gc-digital-talent/pull/11071), creating a mutation and generating a document asynchronously that matches [the current fronted document](https://github.com/GCTC-NTGC/gc-digital-talent/blob/main/apps/web/src/pages/PoolCandidates/ViewPoolCandidatePage/components/ApplicationPrintButton/ApplicationPrintDocument.tsx).

Maybe also replace the singular print items so we  are no longer generating any documents on the frontend? :thinking: 

## 🌎 Localization

All localization should exist in the front end, you just need to move missing ones to the server files.

## ✅ Acceptance Criteria

- [ ] Mutation created for application document download
- [ ] Application document generator created
- [ ] Front end document generators removed and replaced with new mutation
- [ ] Event is fired, queued in job and notification is sent to the appropriate user

### PR DESCRIPTION
**Pull request by esizer**
🤖 Resolves #11126 

## 👋 Introduction

Updates the application document to be asynchronous.

## 🧪 Testing

1. Build the app `pnpm run dev:fresh`
2. Start the queue `make queue-work`
3. Login as admin `admin@test.com`
4. Navigate to a pool candidate
5. Download the application
6. Confirm it works as expected

### CODE REVIEW

**Comment by vd1992** [comment]
An aside, removing all the massive profile/CSV fragments really brought the size down

image

**Comment by vd1992** [comment]
I don't think it was always like this? Was it?

image

**Review by vd1992 — CHANGES_REQUESTED** [review]
The big spacing I think we saw last PR, I think we squeezed things a bit together on the profile download right?

image

And

Clicking this

image

Results in 

image

**Comment by esizer** [comment]
> The big spacing I think we saw last PR, I think we squeezed things a bit together on the profile download right?

Oh shoot, I forgot to use the traits generator :grimacing: Should be fixed after [consolidate experience rendering](https://github.com/GCTC-NTGC/gc-digital-talent/pull/11193/commits/5ef051fcb2e6250715ca81075c7f2cd0937bb3bc)

> Results in

Actually passing in the userId instead of using pool candidate ID would help :woman_facepalming: [fix profile download for candidates](https://github.com/GCTC-NTGC/gc-digital-talent/pull/11193/commits/3403cfc8b610842c064473cf8d7048a424e1f0f7)

**Comment by vd1992** [comment]
Profile download works but I am getting an error for application downloading now

image

image

**Comment by esizer** [comment]
> Profile download works but I am getting an error for application downloading now

Thanks for helping me troubleshoot this, should be fixed after [allow disabling skill generation](https://github.com/GCTC-NTGC/gc-digital-talent/pull/11193/commits/a3ef744686e12650a4643477807d8b995abc1ec0) :woman_shrugging:

File: api/app/Models/User.php [hunk 1]
```diff
@@ -1134,4 +1136,53 @@ public function getIsEmailVerifiedAttribute()
...
+            'written_level' => ['comprehensionLevel', true],
+            'verbal_level' => ['verbalLevel', true],
+            'estimated_language_ability' => ['estimatedLanguageAbility', true],
+            'is_gov_employee' => ['isGovEmployee'],
+            'gov_employee_type' => ['govEmployeeType', true],
+            'has_priority_entitlement' => ['hasPriorityEntitelment'],
+            'priority_number' => ['priorityNumber'],
+            'location_preferences' => ['locationPreferneces', true],
+            'location_exemptions' => ['locationExcemptions'],
+            'accepted_operational_requirements' => ['acceptedOperationalRequirements', true],
```
Comment (vd1992) [inline api/app/Models/User.php:1172]: Oh, some typos

image

**Review by vd1992 — CHANGES_REQUESTED** [review]
Almost there, just your favourite thing outstanding

File: api/app/Models/User.php (hunk 1 above)
Comment (esizer) [reply api/app/Models/User.php]: oops, should be fixed in [fix typos](https://github.com/GCTC-NTGC/gc-digital-talent/pull/11193/commits/3f2a1078ec38d01ac4bfeb6c608b82d9eb1792a1)

**Review by vd1992 — APPROVED** [review]
I think this is in a good position now, great!

**Pull request merged**